### PR TITLE
Let drop column family mutual exclusion with other operations

### PR DESCRIPTION
**Pull request by Linary**
* use ReadWriteLock

Change-Id: Id68401b568d7de8910026fcf420947d42bf23802

### CODE REVIEW

File: hugegraph-rocksdb/src/main/java/com/baidu/hugegraph/backend/store/rocksdb/RocksDBStdSessions.java [hunk 1]
```diff
@@ -247,6 +251,36 @@ public synchronized void dropTable(String... tables)
         }
     }
 
+    public void reload() throws RocksDBException {
```
Comment (javeme) [inline hugegraph-rocksdb/src/main/java/com/baidu/hugegraph/backend/store/rocksdb/RocksDBStdSessions.java:254]: unneeded to reload rocksdb if not operate batch after CF is dropped.
but we can keep these code for future use (don't --amend)

File: hugegraph-rocksdb/src/main/java/com/baidu/hugegraph/backend/store/rocksdb/RocksDBStdSessions.java [hunk 2]
```diff
@@ -67,17 +67,18 @@
...
 import com.google.common.collect.ImmutableList;
 
 public class RocksDBStdSessions extends RocksDBSessions {
 
+    private final HugeConfig config;
     private final String dataPath;
     private final String walPath;
 
-    private final RocksDB rocksdb;
+    private volatile RocksDB rocksdb;
```
Comment (javeme) [inline hugegraph-rocksdb/src/main/java/com/baidu/hugegraph/backend/store/rocksdb/RocksDBStdSessions.java:81]: set final

File: hugegraph-rocksdb/src/main/java/com/baidu/hugegraph/backend/store/rocksdb/RocksDBStore.java [hunk 3]
```diff
@@ -79,6 +83,8 @@
     private RocksDBSessions sessions;
     private final Map<HugeType, String> tableDiskMapping;
 
+    private final ReadWriteLock lock;
```
Comment (javeme) [inline hugegraph-rocksdb/src/main/java/com/baidu/hugegraph/backend/store/rocksdb/RocksDBStore.java:85]: rename to truncateLock

File: hugegraph-rocksdb/src/main/java/com/baidu/hugegraph/backend/store/rocksdb/RocksDBStore.java [hunk 4]
```diff
@@ -344,18 +351,24 @@ public boolean opened() {
...
 
-        for (HugeType type : mutation.types()) {
-            Session session = this.session(type);
-            for (Iterator<BackendAction> it = mutation.mutation(type);
-                 it.hasNext();) {
-                this.mutate(session, it.next());
+            for (HugeType type : mutation.types()) {
+                Session session = this.session(type);
+                for (Iterator<BackendAction> it = mutation.mutation(type);
+                     it.hasNext(); ) {
```
Comment (javeme) [inline hugegraph-rocksdb/src/main/java/com/baidu/hugegraph/backend/store/rocksdb/RocksDBStore.java:365]: remove space

**Review by javeme — APPROVED** [review]

**Review by zhoney — APPROVED** [review]

File: hugegraph-rocksdb/src/main/java/com/baidu/hugegraph/backend/store/rocksdb/RocksDBStdSessions.java [hunk 5]
```diff
@@ -87,6 +88,7 @@ public RocksDBStdSessions(HugeConfig config, String database, String store,
                               String dataPath, String walPath)
                               throws RocksDBException {
         super(config, database, store);
+        this.config = config;
```
Comment (houzhizhen) [inline hugegraph-rocksdb/src/main/java/com/baidu/hugegraph/backend/store/rocksdb/RocksDBStdSessions.java:91]: Config can be removed by making config protected in super class.

File: hugegraph-rocksdb/src/main/java/com/baidu/hugegraph/backend/store/rocksdb/RocksDBStdSessions.java [hunk 6]
```diff
@@ -112,6 +114,7 @@ public RocksDBStdSessions(HugeConfig config, String database, String store,
                               String dataPath, String walPath,
                               List<String> cfNames) throws RocksDBException {
         super(config, database, store);
```
Comment (houzhizhen) [inline hugegraph-rocksdb/src/main/java/com/baidu/hugegraph/backend/store/rocksdb/RocksDBStdSessions.java:116]: using constructor chain by this(...) can reduce some duplicate codes.

**Review by houzhizhen — APPROVED** [review]

**Pull request merged**